tests/Unit/Evolution/Systems/ScalarWave/Test_Equations.cpp

```diff
@@ -50,48 +65,75 @@ void check_du_dt(const size_t npts, const double time) {
+        Tags::deriv<ScalarWave::Psi, tmpl::size_t<Dim>, Frame::Inertial>,
+        Tags::deriv<ScalarWave::Phi<Dim>, tmpl::size_t<Dim>, Frame::Inertial>>>(
+        Scalar<DataVector>(pow<Dim>(npts), gamma2),
+        Scalar<DataVector>(pow<Dim>(npts), 0.0),
+        tnsr::i<DataVector, Dim, Frame::Inertial>(pow<Dim>(npts), 0.0),
+        Scalar<DataVector>(pow<Dim>(npts), 0.0),
+        Scalar<DataVector>(-1.0 * solution.dpsi_dt(x, time).get()),
+        add_scalar_to_tensor_components(solution.dpsi_dx(x, time),
+                                        new_constraint),
+        [&x, &time, &solution ]() noexcept {
+          auto dpi_dx = solution.d2psi_dtdx(x, time);
+          for (size_t i = 0; i < Dim; ++i) {
+            dpi_dx.get(i) *= -1.0;
+          }
+          return dpi_dx;
+        }(),
+        solution.dpsi_dx(x, time), [&npts, &x, &time, &solution ]() noexcept {
+          tnsr::ij<DataVector, Dim, Frame::Inertial> d2psi_dxdx{pow<Dim>(npts),
+                                                                0.0};
+          const tnsr::ii<DataVector, Dim, Frame::Inertial> ddpsi_soln =
+              solution.d2psi_dxdx(x, time);
+          for (size_t i = 0; i < Dim; ++i) {
+            for (size_t j = 0; j < Dim; ++j) {
+              d2psi_dxdx.get(i, j) = ddpsi_soln.get(i, j);
+            }
           }
-        }
-        return d2psi_dxdx;
-      }());
-
-  db::mutate_apply<
-      tmpl::list<Tags::dt<ScalarWave::Pi>, Tags::dt<ScalarWave::Phi<Dim>>,
-                 Tags::dt<ScalarWave::Psi>>,
-      typename ScalarWave::ComputeDuDt<Dim>::argument_tags>(
-      ScalarWave::ComputeDuDt<Dim>{}, make_not_null(&box));
-
-  CHECK_ITERABLE_APPROX(
-      db::get<Tags::dt<ScalarWave::Pi>>(box),
-      Scalar<DataVector>(-1.0 * solution.d2psi_dt2(x, time).get()));
-  CHECK_ITERABLE_APPROX(db::get<Tags::dt<ScalarWave::Phi<Dim>>>(box),
-                        solution.d2psi_dtdx(x, time));
-  CHECK_ITERABLE_APPROX(db::get<Tags::dt<ScalarWave::Psi>>(box),
-                        solution.dpsi_dt(x, time));
+          return d2psi_dxdx;
+        }());
+
+    db::mutate_apply<
+        tmpl::list<Tags::dt<ScalarWave::Pi>, Tags::dt<ScalarWave::Phi<Dim>>,
+                   Tags::dt<ScalarWave::Psi>>,
+        typename ScalarWave::ComputeDuDt<Dim>::argument_tags>(
+        ScalarWave::ComputeDuDt<Dim>{}, make_not_null(&box));
+
+    CHECK_ITERABLE_APPROX(
+        db::get<Tags::dt<ScalarWave::Pi>>(box),
+        Scalar<DataVector>(-1.0 * solution.d2psi_dt2(x, time).get()));
+    CHECK_ITERABLE_APPROX(
+        db::get<Tags::dt<ScalarWave::Phi<Dim>>>(box),
+        add_scalar_to_tensor_components(solution.d2psi_dtdx(x, time),
+                                        -gamma2 * new_constraint));
+    CHECK_ITERABLE_APPROX(db::get<Tags::dt<ScalarWave::Psi>>(box),
+                          solution.dpsi_dt(x, time));
+  };
+
+  // Test with constraint damping parameter set to zero
+  local_check_du_dt(0, 0);
```

Comment:
Here and in the similar lines below, the arguments should end with `.0` (You've used `0.0` style elsewhere in this file)

Reply:
changed